syntax = "proto3";

package yandex.cloud.audit.network;

import "google/protobuf/struct.proto";
import "google/protobuf/timestamp.proto";
import "google/rpc/status.proto";
import "yandex/cloud/audit/common.proto";
import "yandex/cloud/validation.proto";
import "yandex/cloud/vpc/v1/subnet.proto";

option go_package = "a.yandex-team.ru/cloud/bitbucket/public-api/yandex/cloud/audit/vpc;network";
option java_package = "yandex.cloud.api.audit.network";
option java_outer_classname = "SubnetEvents";

message CreateSubnet {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];
    google.protobuf.Struct request_parameters = 12;
    google.protobuf.Struct response = 13;

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
        CANCELLED = 4;
    }

    message EventDetails {
        string subnet_id = 1 [(required) = true];
        string subnet_name = 2;
        string network_id = 3 [(required) = true];
        string network_name = 4;
        string zone_id = 5 [(required) = true];
        repeated string v4_cidr_blocks = 6;
        repeated string v6_cidr_blocks = 7;
        string route_table_id = 8;
        string route_table_name = 9;
        yandex.cloud.vpc.v1.DhcpOptions dhcp_options = 10;
    }
}

message UpdateSubnet {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];
    google.protobuf.Struct request_parameters = 12;
    google.protobuf.Struct response = 13;

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
    }

    message EventDetails {
        string subnet_id = 1 [(required) = true];
        string subnet_name = 2;
        string network_id = 3 [(required) = true];
        string network_name = 4;
        string zone_id = 5 [(required) = true];
        repeated string v4_cidr_blocks = 6;
        repeated string v6_cidr_blocks = 7;
        string route_table_id = 8;
        string route_table_name = 9;
        yandex.cloud.vpc.v1.DhcpOptions dhcp_options = 10;
   }
}

message DeleteSubnet {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];
    google.protobuf.Struct request_parameters = 12;
    google.protobuf.Struct response = 13;

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
    }

    message EventDetails {
        string subnet_id = 1 [(required) = true];
        string subnet_name = 2;
        string network_id = 3 [(required) = true];
        string network_name = 4;
    }
}

message MoveSubnet {
    string event_id = 1 [(required) = true];
    string event_source = 2 [(required) = true];
    string event_type = 3 [(required) = true];
    google.protobuf.Timestamp event_time = 4 [(required) = true];
    Authentication authentication = 5 [(required) = true];
    Authorization authorization = 6 [(required) = true];
    ResourceMetadata resource_metadata = 7 [(required) = true];
    RequestMetadata request_metadata = 8 [(required) = true];
    EventStatus event_status = 9 [(required) = true];
    google.rpc.Status error = 10;
    EventDetails details = 11 [(required) = true];

    enum EventStatus {
        EVENT_STATUS_UNSPECIFIED = 0;
        STARTED = 1;
        ERROR = 2;
        DONE = 3;
    }

    message EventDetails {
        string subnet_id = 1 [(required) = true];
        string subnet_name = 2;
        string network_id = 3 [(required) = true];
        string network_name = 4;
    }
}
